src/daft-logical-plan/src/ops/monotonically_increasing_id.rs

```diff
@@ -17,7 +17,8 @@ impl MonotonicallyIncreasingId {
         let column_name = column_name.unwrap_or("id");
 
         let mut schema_with_id_index_map = input.schema().fields.clone();
-        schema_with_id_index_map.insert(
+        schema_with_id_index_map.shift_insert(
+            0,
```

Comment:
Why? Seems like an orthogonal change to the rest of the PR

Reply:
Ok this is a long explanation but it's necessary.

One discrepancy between the native and python readers I found is that if you pass in a schema that reorders the columns, the native reader does not actually respect that. The reason turned out to be because an unloaded micropartition only creates a new scan task if the schemas differ ([see here](https://github.com/Eventual-Inc/Daft/blob/kevin/bye-py-storage-config/src/daft-micropartition/src/ops/cast_to_schema.rs#L22)). However, the impl of Eq for Schema relies on Eq for `indexmap::IndexMap`, which does not actually check for ordering. Thus the schemas are erronenously presumed to be equal.

So, I made the equality check for Schema stricter, however that caused an issue with monotonically increasing ID, since the schema built in the logical plan (which has the ID at the end) was different from the schema of the actual computed tables (which have the ID as the first column). That's why I need to `shift_insert` instead of `insert` here.

Reply:
Whaaaat 😮 

Ok can we lift this to the PR description? Very surprised that this was surfaced only now though, seems like a bug that should have come up sooner.